test/chart.test.ts

```diff
@@ -15,6 +15,38 @@ test('empty stack', () => {
   expect(Testing.synth(chart)).toMatchSnapshot();
 });
 
+test('disabling resource name hashes at chart level', () => {
+  // GIVEN
+  const app = Testing.app();
+  const chart = new Chart(app, 'test', {
+    enableResourceNameHashes: false,
+  });
+
+  // WHEN
+  const ob1 = new ApiObject(chart, 'resource1', { kind: 'Resource1', apiVersion: 'v1' });
+  const ob2 = new ApiObject(chart, 'resource2', { kind: 'Resource3', apiVersion: 'v1' });
+
+  // THEN
+  expect(ob1.name).toEqual('test-resource1');
+  expect(ob2.name).toEqual('test-resource2');
+  expect(Testing.synth(chart)).toMatchSnapshot();
+});
+
+test('resource name hashes work by default', () => {
+  // GIVEN
+  const app = Testing.app();
+  const chart = new Chart(app, 'test');
+
+  // WHEN
+  const ob1 = new ApiObject(chart, 'resource1', { kind: 'Resource1', apiVersion: 'v1' });
+  const ob2 = new ApiObject(chart, 'resource2', { kind: 'Resource3', apiVersion: 'v1' });
+
+  // THEN
+  expect(ob1.name).not.toEqual('test-resouce1');
```

Comment:
Better to assert the positive instead of the negative, since the negative would make this test pass even if the name is completely different than what we expect. 

```suggestion
  expect(ob1.name).toEqual('test-resouce1');
```